pkg/tasks/update_template_distributions.go

```diff
@@ -0,0 +1,200 @@
+	if t.payload.RepoConfigUUIDs == nil {
+		return nil
+	}
+
+	repoConfigDistributionHref := map[string]string{} // mapping to associate each repo config to a distribution href
+
+	reposAdded, reposRemoved, reposUnchanged, allRepos, err := t.daoReg.Template.GetRepoChanges(t.payload.TemplateUUID, t.payload.RepoConfigUUIDs)
+	if err != nil {
+		return err
+	}
+
+	l := api.ListSnapshotByDateRequest{Date: t.payload.TemplateDate, RepositoryUUIDS: allRepos}
+	snapshots, err := t.daoReg.Snapshot.FetchSnapshotsModelByDateAndRepository(t.orgId, l)
+	if err != nil {
+		return err
+	}
+
+	for _, repoConfigUUID := range allRepos {
+		repo, err := t.daoReg.RepositoryConfig.Fetch(t.orgId, repoConfigUUID)
+		if err != nil {
+			return err
+		}
+
+		var distributionPath string
+		if repo.OrgID == config.RedHatOrg {
+			path, err := getRHRepoContentPath(repo.URL)
+			if err != nil {
+				return err
+			}
+			t.pulpClient = t.pulpClient.WithDomain(t.rhDomainName)
+			distributionPath = fmt.Sprintf("templates/%v/%v", t.payload.TemplateUUID, path)
+		} else {
+			t.pulpClient = t.pulpClient.WithDomain(t.domainName)
+			distributionPath = fmt.Sprintf("templates/%v/%v", t.payload.TemplateUUID, repoConfigUUID)
+		}
+
+		snapIndex := slices.IndexFunc(snapshots, func(c models.Snapshot) bool {
+			return c.RepositoryConfigurationUUID == repoConfigUUID
+		})
+
+		distName := t.payload.TemplateUUID + "/" + snapshots[snapIndex].UUID
+
+		if reposAdded != nil && slices.Contains(reposAdded, repoConfigUUID) {
+			var contentGuardHref *string
+			if t.orgId != config.RedHatOrg && config.Get().Clients.Pulp.CustomRepoContentGuards {
+				href, err := t.pulpClient.CreateOrUpdateGuardsForOrg(t.orgId)
+				if err != nil {
+					return fmt.Errorf("could not fetch/create/update content guard: %w", err)
+				}
+				contentGuardHref = &href
+			}
+			distTask, err := t.pulpClient.CreateRpmDistribution(snapshots[snapIndex].PublicationHref, distName, distributionPath, contentGuardHref)
+			if err != nil {
+				return err
+			}
+
+			distResp, err := t.pulpClient.PollTask(*distTask)
+			if err != nil {
+				return err
+			}
+
+			distHrefPtr := pulp_client.SelectRpmDistributionHref(distResp)
+			if distHrefPtr == nil {
+				return fmt.Errorf("Could not find a distribution href in task: %v", *distResp.PulpHref)
+			}
+
+			repoConfigDistributionHref[repoConfigUUID] = *distHrefPtr
+		}
+
+		if reposRemoved != nil && slices.Contains(reposRemoved, repoConfigUUID) {
+			distHref, err := t.daoReg.Template.GetDistributionHref(t.payload.TemplateUUID, repoConfigUUID)
+			if err != nil {
+				return err
+			}
+			taskHref, err := t.pulpClient.DeleteRpmDistribution(distHref)
+			if err != nil {
+				return err
+			}
+			_, err = t.pulpClient.PollTask(taskHref)
+			if err != nil {
+				return err
+			}
+
+			keepRepoConfigUUIDs := append(reposUnchanged, reposAdded...)
+			err = t.daoReg.Template.DeleteTemplateRepoConfigs(t.payload.TemplateUUID, keepRepoConfigUUIDs)
```

Comment:
This is run once per repoConfig (since its in the `for _, repoConfigUUID := range allRepos {` loop , but seems like it probably only needs to be run once at the end?

Reply:
because it deletes all repo configs that aren't in the 'keep list'